specifyweb/frontend/js_src/lib/components/Molecules/Dialog.tsx

```diff
@@ -364,33 +361,42 @@ export function Dialog({
     );
   }, [showIcon, defaultIcon, buttons, buttonContainer]);
 
-  const overlayElement: Props['overlayElement'] = React.useCallback(
-    (
-      props: React.ComponentPropsWithRef<'div'>,
-      contentElement: React.ReactElement
-    ) => (
-      <div
-        {...props}
-        onMouseDown={
-          closeOnOutsideClick
-            ? (event): void => {
-                // Outside click detection
-                if (
-                  modal &&
-                  typeof handleClose === 'function' &&
-                  event.target === event.currentTarget
-                ) {
-                  event.preventDefault();
-                  handleClose();
-                } else props?.onMouseDown?.(event);
-              }
-            : undefined
-        }
-      >
-        {contentElement}
-      </div>
-    ),
-    [modal, handleClose, closeOnOutsideClick]
+  const allowCloseRef = React.useRef<boolean>(false);
```

Comment:
and the feature is more about "preventing accidental clicks" rather than "allowing close", so could you please rename the variable to something like "ignoreOutsideClickRef" or etc to be more self-descriptive